### PR TITLE
Adding start_mode and rapid_fail_protection app pool advanced settings

### PR DESCRIPTION
**Pull request by brushwood24**
Added start_mode and rapid_fail_protection.  I also reorganized the parameters to be on separate lines as mentioned in issue 63 to make it a little easier to read.  Let me know if I should separate this out into a different commit.

### CODE REVIEW

File: manifests/manage_app_pool.pp [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+define iis::manage_app_pool(
+  $app_pool_name = $title,
+  $enable_32_bit = false,
+  $managed_runtime_version = 'v4.0',
+  $managed_pipeline_mode = 'Integrated',
+  $ensure = 'present',
+  $start_mode = 'OnDemand',
+  $rapid_fail_protection = true
+)
+{
```
Comment (igalic) [inline manifests/manage_app_pool.pp]: these two should be on the same line.

File: manifests/manage_app_pool.pp [hunk 2]
```diff
@@ -1,10 +1,21 @@
 #
-define iis::manage_app_pool($app_pool_name = $title, $enable_32_bit = false, $managed_runtime_version = 'v4.0', $managed_pipeline_mode = 'Integrated', $ensure = 'present') {
+define iis::manage_app_pool(
+  $app_pool_name = $title,
+  $enable_32_bit = false,
+  $managed_runtime_version = 'v4.0',
+  $managed_pipeline_mode = 'Integrated',
+  $ensure = 'present',
+  $start_mode = 'OnDemand',
+  $rapid_fail_protection = true
```
Comment (igalic) [inline manifests/manage_app_pool.pp]: = should be aligned

File: manifests/manage_app_pool.pp [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
+)
+{
 
   validate_bool($enable_32_bit)
   validate_re($managed_runtime_version, ['^(v2\.0|v4\.0)$'])
   validate_re($managed_pipeline_mode, ['^(Integrated|Classic)$'])
   validate_re($ensure, '^(present|installed|absent|purged)$', 'ensure must be one of \'present\', \'installed\', \'absent\', \'purged\'')
+  validate_re($start_mode,['^(OnDemand|AlwaysRunning)$'])
```
Comment (igalic) [inline manifests/manage_app_pool.pp]: this should be either:
```puppet
validate_re($start_mode, '^(OnDemand|AlwaysRunning)$')
```
or
```puppet
validate_re($start_mode, ['^OnDemand$, '^AlwaysRunning$'])
```

but not a combination seems weird. (this particular line is lacking a space after the comma, too)


I suggest aligning it all to one style. (i prefer the first)

**Comment by igalic** [comment]
could you please verify if powershell will permit `Foo:/bar/baz` syntax instead of `Foo:\bar\baz`?
the latter needs to be escaped, a lot, and looks completely unreadable.

**Comment by brushwood24** [comment]
Powershell will permit Foo:/bar/baz instead of Foo:\bar\baz.  Since I was making an addition I just stayed with the current style to keep it consistent but I can change it.

**Comment by igalic** [comment]
@brushwood24 i'd be happy with a follow-up pr for that

**Pull request merged**